tock/hours/models.py

```diff
@@ -349,6 +355,16 @@ class TimecardNote(models.Model):
         default=True,
         help_text='Toggle whether or not the note is displayed in a timecard.'
     )
+    display_period_start = models.DateField(
+        null=True,
+        blank=True,
+        help_text='The start date for displaying this note. Note: manually enabling will override this setting.'
+    )
+    display_period_end = models.DateField(
```

Comment:
What if you wanted to start a note on May 1 and leave it up for an undetermined amount of time? How would that work?

Reply:
Right - this implementation doesn't support that, as it requires you to specify either both the start and end date, or neither. It would be pretty straightforward to modify this to support that use case if that's something we want to do.

Reply:
I suppose one could just put in a far-future end date and call it good, too.